File: Individual-Sorting-Methods/C#-Versions/SelectionSort.cs
using System;

namespace SelectionSort
{
    class Program
    {
        static void Main(string[] args)
        {
            int[] arr = {3, 2, 5, 1, 4, 10, 9, 6, 8, 7, 13, 12, 11};

            Console.WriteLine("Unsorted: ");
            Display(arr);
            Selection(arr);
            Console.ReadKey();

            Console.WriteLine("");
            Console.WriteLine("Sorted: ");
            Display(arr);
            Console.WriteLine("");

            Console.ReadKey();
        }

        static void Display(int[] A)
        {
            Console.WriteLine("");

            for(int i = 0; i < A.Length; i++)
            {
                if(i == A.Length - 1)
                {
                    Console.Write(A[i] + "");
                    Console.WriteLine("");
                    return;
                }

                Console.Write(A[i] + ", ");
            }

            Console.WriteLine("");
        }

        static void Selection(int[] A)
        {
            int j, k;
            for(int i = 0; i < A.Length - 1; i++)
            {
                k = i;
                for(j = i + 1; j < A.Length; j++)
                {
                    if(A[j] < A[k])
                        k = j;
                }

                Swap(A, k, i);
            }
        }

        static void Swap(int[] A, int i, int j)
        {
            int tmp = A[i];
            A[i] = A[j];
            A[j] = tmp;
        }
    }
}

File: Individual-Sorting-Methods/Java-Versions/selectionSort.java
public class selectionSort
{
    public static void main(String args[])
    {
        int arr[] = {3, 2, 5, 1, 4, 10, 9, 6, 8, 7, 13, 12, 11};

        System.out.println("Unsorted: ");
        display(arr);
        selection(arr);
        System.out.println("");
        System.out.println("Sorted: ");
        display(arr);
        System.out.println("");
    }

    public static void display(int A[])
    {
        System.out.println("");
        
        for(int j = 0; j < A.length; j++)
        {
            if(j == A.length - 1)
            {
                System.out.print(A[j] + "");
                System.out.println("");
                return;
            }

            System.out.print(A[j] + ", ");
        }

        System.out.println("");
    }

    public static void selection(int A[])
    {
        int j, k;
        for(int i = 0; i < A.length - 1; i++)
        {
            k = i;
            for(j = i + 1; j < A.length; j++)
            {
                if(A[j] < A[k])
                    k = j;
            }

            swap(A, k, i);
        }
    }

    public static void swap(int A[], int i, int j)
    {
        int tmp = A[i];
        A[i] = A[j];
        A[j] = tmp;
    }
}